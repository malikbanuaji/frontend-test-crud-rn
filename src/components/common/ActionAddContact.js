import React from 'react';
import {StyleSheet, Text, View, TouchableNativeFeedback} from 'react-native';
import {Mixins, Spacing, Colors, Typography} from '../../styles';
import Icon from 'react-native-vector-icons/Feather';

export default function ActionAddContact({onPress}) {
  return (
    <View style={styles.actionAddContact}>
      <TouchableNativeFeedback
        testID={'action-add-contact-button'}
        onPress={onPress}>
        <View style={styles.actionAddContactButton}>
          <Icon key={'icon'} name={'plus'} size={24} style={styles.icon} />
          <Text style={styles.caption}>{'Tambah kontak'}</Text>
        </View>
      </TouchableNativeFeedback>
    </View>
  );
}

const styles = StyleSheet.create({
  actionAddContact: {
    ...Mixins.baseRounded,
    overflow: 'hidden',
  },
  actionAddContactButton: {
    ...Mixins.baseRounded,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: Spacing.small,
    backgroundColor: Colors.primary,
  },
  caption: {
    ...Typography.actionButtonText,
  },
  icon: {
    marginRight: Spacing.smallest,
  },
});
