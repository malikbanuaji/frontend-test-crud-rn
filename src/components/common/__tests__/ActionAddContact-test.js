import React from 'react';

import renderer from 'react-test-renderer';
import ActionAddContact from '../ActionAddContact';

test('renders correctly', () => {
  const tree = renderer.create(<ActionAddContact />).toJSON();
  expect(tree).toMatchSnapshot();
});
