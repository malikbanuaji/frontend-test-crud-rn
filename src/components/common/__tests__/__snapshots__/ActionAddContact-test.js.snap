// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#93b5e1",
        "borderRadius": 8,
        "display": "flex",
        "flexDirection": "row",
        "overflow": "hidden",
        "padding": 12,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "marginRight": 4,
          },
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï†´
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Rubik-Medium",
          "fontSize": 14,
        }
      }
    >
      Tambah kontak
    </Text>
  </View>
</View>
`;
